pkg/protocol/xprotocol/bolt/protocol.go

```diff
@@ -224,5 +224,6 @@ func (proto boltProtocol) EnableWorkerPool() bool {
 }
 
 func (proto boltProtocol) GenerateRequestID(streamID *uint64) uint64 {
-	return atomic.AddUint64(streamID, 1)
+	// fix bug for issue: https://github.com/mosn/mosn/issues/2403
+	return uint64(uint32(atomic.AddUint64(streamID, 1)))
```

Comment:
I wonder why not add the conversion outside the GenerateRequestID? So we don't need to do it in each protocol.

Reply:
> I wonder why not add the conversion outside the GenerateRequestID? So we don't need to do it in each protocol.

some protocol‘s request id is uint64

Reply:
Got it. It is a protocol-specific boxing.